plugins/database/postgresql/postgresql_test.go

```diff
@@ -213,128 +283,251 @@ func TestPostgreSQL_RenewUser(t *testing.T) {
-			},
+		"default statements with name": {
+			initialExpiration:  now.Add(1 * time.Minute),
+			newExpiration:      now.Add(5 * time.Minute),
+			expectedExpiration: now.Add(5 * time.Minute),
+			statements:         []string{defaultExpirationStatement},
+			expectErr:          false,
 		},
-		"default username": {
-			statements: []string{defaultPostgresRotateRootCredentialsSQL},
+		"default statements with username": {
+			initialExpiration:  now.Add(1 * time.Minute),
+			newExpiration:      now.Add(5 * time.Minute),
+			expectedExpiration: now.Add(5 * time.Minute),
+			statements:         []string{`ALTER ROLE "{{username}}" VALID UNTIL '{{expiration}}';`},
+			expectErr:          false,
+		},
+		"bad statements": {
+			initialExpiration:  now.Add(1 * time.Minute),
+			newExpiration:      now.Add(5 * time.Minute),
+			expectedExpiration: now.Add(1 * time.Minute),
+			statements:         []string{"ladshfouay09sgj"},
+			expectErr:          true,
 		},
 	}
 
+	// Shared test container for speed - there should not be any overlap between the tests
+	db, cleanup := getPostgreSQL(t, nil)
+	defer cleanup()
+
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			cleanup, connURL := postgresql.PrepareTestContainer(t, "latest")
-			defer cleanup()
-			connURL = strings.Replace(connURL, "postgres:secret", `{{username}}:{{password}}`, -1)
-
-			connectionDetails := map[string]interface{}{
-				"connection_url":       connURL,
-				"max_open_connections": 5,
-				"username":             "postgres",
-				"password":             "secret",
+			password := "myreallysecurepassword"
+			initialExpiration := test.initialExpiration.Truncate(time.Second)
+			createReq := newdbplugin.NewUserRequest{
+				UsernameConfig: newdbplugin.UsernameMetadata{
+					DisplayName: "test",
+					RoleName:    "test",
+				},
+				Statements: newdbplugin.Statements{
+					Commands: []string{createAdminUser},
+				},
+				Password:   password,
+				Expiration: initialExpiration,
 			}
+			createResp := dbtesting.AssertNewUser(t, db, createReq)
 
-			db := new()
-			connProducer := db.SQLConnectionProducer
+			assertCredsExist(t, db.ConnectionURL, createResp.Username, password)
 
-			// Give a timeout just in case the test decides to be problematic
-			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-			defer cancel()
-
-			_, err := db.Init(ctx, connectionDetails, true)
-			if err != nil {
-				t.Fatalf("err: %s", err)
+			actualExpiration := getExpiration(t, db, createResp.Username)
+			if actualExpiration.IsZero() {
+				t.Fatalf("Initial expiration is zero but should be set")
+			}
+			if !actualExpiration.Equal(initialExpiration) {
+				t.Fatalf("Actual expiration: %s Expected expiration: %s", actualExpiration, initialExpiration)
 			}
 
-			if !connProducer.Initialized {
-				t.Fatal("Database should be initialized")
+			newExpiration := test.newExpiration.Truncate(time.Second)
```

Comment:
Should we truncate this value to the second so that we don't fail due to precision?